python-lib/results/ab_statistics.py

```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+
+from constants import Group, Columns
+
+
+class AbStatistics(object):
+    def __init__(self, user_reference_column: str, group_column: str, conversion_column: str):
+        self.user_reference_column = user_reference_column
+        self.group_column = group_column
+        self.conversion_column = conversion_column
+
+    def compute(self, results_df):
+        results_df = self.check_results_df(results_df)
+        aggregation = self.aggregate_by_group(results_df)
+        statistics_df = self.format_statistics(aggregation)
+        return statistics_df
+
+    def check_results_df(self, result_df):
+        valid_rows = result_df[[self.user_reference_column, self.group_column, self.conversion_column]].dropna()
+        not_empty_rows_nb = valid_rows.shape[0]
+        if not result_df[self.user_reference_column].is_unique:
+            raise ValueError("There should be only one row per user in the input dataset")
+        if not_empty_rows_nb < 2:
+            raise ValueError("The input dataset should contain at least 2 users with conversion and group references")
+        invalid_groups = valid_rows[(valid_rows[self.group_column] != Group.A.value) & (valid_rows[self.group_column] != Group.B.value)]
+        if not invalid_groups.empty:
+            raise ValueError("The group indicator should be either 'A' or 'B'")
+        invalid_conversion = valid_rows[(valid_rows[self.conversion_column] != 0) & (valid_rows[self.conversion_column] != 1)]
+        if not invalid_conversion.empty:
+            raise ValueError("The group indicator should be either 0 or 1")
```

Comment:
copy paste mistake

Reply:
done in https://github.com/dataiku/dss-plugin-ab-testing/pull/2/commits/c9888533223d12828b49e2a8ca56a4030fb93a6d